cms/views.py

```diff
@@ -162,6 +162,10 @@ def details(request, slug):
             toolbar.redirect_url = redirect_url
         elif redirect_url not in own_urls:
             # prevent redirect to self
+            if getattr(settings, 'REDIRECT_PRESERVE_QUERY_PARAMS', False):
```

Comment:
Could you change this to use `get_cms_setting` please.

It'll need the default adding here; https://github.com/django-cms/django-cms/blob/develop/cms/utils/conf.py#L43

And this will mean that the setting will be valid in the `CMS_` namespace. So the setting will be `CMS_REDIRECT_PRESERVE_QUERY_PARAMS`

Reply:
Done. 
Changed to use `get_cms_setting` and added on the `DEFAULTS`.